.readthedocs.yaml

```diff
@@ -0,0 +1,18 @@
+# .readthedocs.yaml
+# Read the Docs configuration file
+# See https://docs.readthedocs.io/en/stable/config-file/v2.html for details
+
+# Required
+version: 2
+
+# Set the version of Python and other tools you might need
+build:
+  os: ubuntu-20.04
+  tools:
+    python: "miniconda3-4.7"
+
+mkdocs:
+  configuration: mkdocs.yml
+
+conda:
+  environment: docs/environment.yml
```

Comment:
> 



> According to the RTD docs, this snippet should work with Poetry and does not require a separate Conda env and requirements. Could you try it out?

I've been doing some experiments with the docs of ClinicaDL. I tried exactly the same snippet in commit [c7e1cc](https://github.com/aramis-lab/clinicadl/pull/320/commits/c7e1ccb473615ddd300087f9ee241db4e6299316) in this project. It seems that RTD ignores the `poetry.lock` file and it installs the dependencies declared in `pyproject.toml` using the pip engine for solving dependencies. As result, it installs  the wrong version of the `MarkUpSafe` package which is incompatible with Jinja2<3.0.0 (and the command`mkdocs build` fails systematically) ([see here the output message of the failed build](https://readthedocs.org/projects/clinicadl/builds/17329331/)).

I could try it out in this project but I have little hope that it will work

Reply:
Well, the modifications seems to work here... don't know why the same code doesn't work in ClinicaDL :s

Reply:
Indeed, here it works because Jinja2 is already higher than 3.0.0!

Reply:
>  It seems that RTD ignores the poetry.lock file and it installs the dependencies declared in pyproject.toml using the pip engine for solving dependencies.

Technically what happens is that RTD runs `pip install -e .[docs]` which works with any PEP 631 compliant build system, which includes Poetry. The exact dependencies in the lock file will not be honoured since `pip install` is used instead of `poetry install` but it's not a big deal for docs.